pkg/registrymirror/registrymirrorsstore.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	for _, opt := range opts {
+		opt(s)
+	}
+
+	s.systemContext = &ciTypes.SystemContext{SystemRegistriesConfPath: s.configPath}
+	return s
+}
+
+// Cleanup resets the store which includes in-memory and disk resources.
+func (s *FileStore) Cleanup() {
+	s.ruleRWMutex.Lock()
+	defer s.ruleRWMutex.Unlock()
+
+	s.icspRules = make(map[types.UID]*operatorV1Alpha1.ImageContentSourcePolicy)
+	s.idmsRules = make(map[types.UID]*configV1.ImageDigestMirrorSet)
+	s.itmsRules = make(map[types.UID]*configV1.ImageTagMirrorSet)
+
+	if err := os.Remove(s.configPath); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to cleanup registries config at %q: %v", s.configPath, err)
+	}
+	sysregistriesv2.InvalidateCache()
+}
+
+func (s *FileStore) updateConfig() error {
+	if s.updateDelay == 0 {
+		return s.updateConfigNow()
+	}
+
+	return s.updateConfigDelayed()
+}
+
+func (s *FileStore) updateConfigDelayed() error {
+	// Using a different mutex to protect timer vs. rules to avoid deadlock.
+	s.timerMutex.Lock()
+	defer s.timerMutex.Unlock()
+
+	if s.timer != nil {
+		s.timer.Stop()
+	}
+
+	s.timer = time.AfterFunc(s.updateDelay, func() {
+		if err := s.updateConfigNow(); err != nil {
+			log.Errorf("Failed to update the registry mirror config: %v", err)
+		}
+	})
+
+	return nil
+}
+
+func (s *FileStore) updateConfigNow() error {
+	icspRules, idmsRules, itmsRules := s.getAllMirrorSets()
+
+	// Populate config.
+	config := new(sysregistriesv2.V2RegistriesConf)
+	err := registries.EditRegistriesConfig(config, nil, nil, icspRules, idmsRules, itmsRules)
+	if err != nil {
+		return errors.Wrap(err, "could not create registries config")
+	}
+
+	// Encode config.
+	var newData bytes.Buffer
+	encoder := toml.NewEncoder(&newData)
+	if err := encoder.Encode(config); err != nil {
+		return errors.Wrap(err, "could not encode registries config")
+	}
+
+	// Ensure output dir exists.
+	err = os.MkdirAll(filepath.Dir(s.configPath), 0755)
+	if err != nil {
+		return errors.Wrap(err, "could not create directories")
+	}
+
+	// Open handle to output file.
+	f, err := os.Create(s.configPath)
+	if err != nil {
+		return errors.Wrap(err, "could not create/open file")
+	}
+	defer utils.IgnoreError(f.Close)
+
+	// Write encoded config to output file.
+	_, err = f.Write(newData.Bytes())
+	if err != nil {
+		return errors.Wrap(err, "could not write bytes to file")
+	}
+
+	// Close file before defer to ensure close errors are surfaced.
+	if err := f.Close(); err != nil {
+		return errors.Wrap(err, "could not close file")
+	}
+
+	// Invalidate the registries cache so that the updated file will be read on next invocation of PullSources.
+	sysregistriesv2.InvalidateCache()
+
+	log.Debugf("Successfully updated the registries config at %q with %v icspRules, %v idmsRules, %v itmsRules",
+		s.configPath, len(icspRules), len(idmsRules), len(itmsRules))
+
+	return nil
+}
+
+// PullSources will return image references in the order they should be attempted when pulling the image.
```

Comment:
nit: this is already documented in the interface definition, and it does not add any implementation-specific details, so you can remove this

Reply:
Our style checks fail if I have nothing, so I copy/pasted - I'll change it to 'see interface' for description.